### DISCUSSION TITLE
New Build - Possibly H2

### DISCUSSION THREAD

**Discussion by Lester444** [kachurovskiy/nanoels#73] [Show and tell]
Greetings.

First, a big thank you to Maxim for making this project available to everyone. The images and videos were a big help to me while I was trying to figure out how to make one.

I have started my build.

The encoder's gear was not 3D printed but rather cut on a CNC router. Material used is flat Acrylic plastic 4.5mm thick. The gear was then glued onto a hub  (Polyester plastic) to hold it on the encoder's shaft.
![Gear small](https://user-images.githubusercontent.com/23709087/194534459-6f36588f-e2e3-40ed-b894-169e2df57df5.jpg)
![Encoder_small](https://user-images.githubusercontent.com/23709087/194534551-4b038b09-d858-45e2-9e6f-90584af61fdb.jpg)

I am using a Servo Stepper motor, NEMA23 size. I was planning on using a belt but got impatient, then I realized I can re-use one original gear from the lathe and the other gear was again CNC cut from 6063-T6 aluminum plate 6.4mm thick. The hub was made on the lathe itself, from 6063-T6 aluminum rod 25mm diameter. The gear and hub were glued together, then a hole was drilled in between the interface of the hub & gear, and 3mm pin was glued into this hole. The pin acts like a shear key but this one can't be replaced.
![Drive gear small](https://user-images.githubusercontent.com/23709087/194534650-6357b0d2-0dcc-42f4-9c97-8b67b45cb3ec.jpg)

The motor mount was cut from 10.5mm-thick aluminum plate.

I plan to use a different layout for the buttons & display but that's just a personal preference. An Acrylic pastic housing is already made but I have not yet drilled holes for the buttons and switches, because some of the switches I do not yet understand how they are used.
![Housing small](https://user-images.githubusercontent.com/23709087/194534734-e81016e9-e097-4d34-9db1-b964efa1e353.jpg)

These buttons I do not understand how to use:
![Function of these buttons](https://user-images.githubusercontent.com/23709087/194534794-d597a7d8-eb6a-400f-84e6-000aa4b0725d.png)

**Comment by Lester444** [comment]
Here's a short video of the encoder to see if it is functional. It turns out I have a 800PPR encoder.
https://youtu.be/WE-92busVM8

**Comment by kachurovskiy** [comment]
@Lester444 looking great! Please keep us updated on the progress. Here's what H2 buttons do:

![image](https://user-images.githubusercontent.com/517919/194547986-f6f26677-896e-4a3b-a0f4-17e472824b2c.png)

**Reply by Lester444** [reply]
Pushbuttons and LCD display are now mounted on the housing.
![Buttons 1](https://user-images.githubusercontent.com/23709087/194679859-f50203ae-3591-4af8-8de8-7136ba20bf19.jpg)

I have also uploaded the H2.ino (with some edits for LCD display & encoder) into Arduino and it seems to be working fine.
![Nano is working](https://user-images.githubusercontent.com/23709087/194679937-ed2f4d3a-879d-4fd0-a2cd-c16d1c4d9c04.jpg)

Next thing to do is to wire everything together while I wait for the power supply to get here.

I hope it all works well! :)

**Comment by Lester444** [comment]
Quick update:

Most switches have been wired (B0~B6).
Arduino Nano has been loaded with H2.ino.
Tacho is running and I get the correct RPM (it turns out I actually have 400PPR encoder).
ON / off button is working (based on the display only).
STOPL is working, seen on the display as "L"
STOPR is working, seen on the display as "R"
Stepper motor is not yet wired.

Issues I am having right now:
I am not able to change the pitch. I pressed "+" or "-" buttons but the pitch remains at 2.0mm.
Pressing "Left" or "Right" buttons, I can see the position changing on the display, but goes back to zero when I release the button. So the pushbutton is fine, but what is wrong then?

I do not understand what is wrong.

**Reply by kachurovskiy** [reply]
Looks like your 2.00mm shortcut button is shorted to the GMD.

**Reply by Lester444** [reply]
> Looks like your 2.00mm shortcut button is shorted to the GMD.

So I added a pull-up resistor to "AB" and the pushbuttons work fine now.

     -  BUT -

Based on the schematic diagram, "AB" is always pulled down to GND by resistor R9. 
![Analog switches](https://user-images.githubusercontent.com/23709087/194701797-cb719e92-db09-4f65-bad6-018bff4b7c5c.png)

How should I wire these "analog" switches?

**Reply by Lester444** [reply]
It's OK now, thank you Maxim.

I left pin "AB" floating. Adding the 1M resistor solved my problem. (Y)

**Comment by Lester444** [comment]
This is how I wired the switches, all of these switches connect to GND when pressed:
![Switches only](https://user-images.githubusercontent.com/23709087/194699696-3c633c4e-3280-49b6-bd26-79c4a6d6113a.png)

**Comment by Lester444** [comment]
It's running! :)

I still need to get comfortable with using this so I'm practicing on wood for now.

https://youtu.be/HJqsfwterUE

**Reply by kachurovskiy** [reply]
Horay! Congratulations 👏

**Comment by Lester444** [comment]
Hello Maxim.

I thought I would document my learnings in this project and some of the changes I made to the hardware and firmware. Perhaps somebody would find it useful in the future?

Stepper Motor:
It is false economy the get the biggest NEMA23.... then scrimp on the power supply. I learned this the hard way. The motor would simply stall. Right now I am using 2 pieces of 19v/5A laptop chargers as power supply, wired in series for 38v going to the motor driver. No more stall. I prefer to use laptop chargers because they are:
	1. Cheap
	2. Current-limited
	3. Fully enclosed so no need for another enclosure
	4. Dust-proof, splash-proof
	5. Easy to buy 

Gear Drive for the leadscrew:
Aluminum gear is very noisy during jogging (moves). In the near future I will change this gear to Acetal plastic to reduce the noise level. But right now I am happy to have this ELS running!


My scary moment:
The pitch sometimes suddenly changes from 0.2mm to 2.0mm and it's very scary!!! I think this is caused by electrical noise in my environment. I added 0.1uF caps on all digital pin switches (B0~B6) as close as I can get to the NANO pin itself. 

AnalogRead (A6): I edited the software so that it no longer uses analogRead(_). Instead, I added switches to NANO pins A0, A1, A2 which are not used anyway. So now I do digitalRead(A0) and it is immune to electrical noise. 

Thank you very much Maxim, your generosity is greatly appreciated. Warm greetings from the Philippines!
  
(BTW I have my eyes on the MEGA ELS....... :)  )

**Reply by kachurovskiy** [reply]
One of the main benefits of using the PCB is that it has a ground plane which eliminates noise problems. I use 1 analog input instead of many because reading many inputs is slower and slightly reduces the maximum stepper RPM the ELS can achieve. The sudden change of pitch is indeed very dangerous and even though I've never experienced it, it would be worthwhile adding precautions against such things in the analog button handling code. Thank you for sharing your experience 👍
